core/src/data/clientGeoJsonSource.cpp

```diff
@@ -20,6 +20,20 @@ const uint8_t indexMaxZoom = 18;
 const uint32_t indexMaxPoints = 100000;
 double tolerance = 1E-8;
 
+class Task : public TileTask {
```

Comment:
Instead of creating this temp `Task` class for `clientGeoJsonSource`, how about not making the base `TileTask` class abstract and make it implement `hasData` method to return `true`.

Other data source classes are already overriding this method.